### PR TITLE
Updated Catch2 to v3

### PR DESCRIPTION
**Pull request by mcparry**
As described in issue #823 .
I am a first-time contributor, so please check carefully :-)

### CODE REVIEW

File: tests/ssl/CMakeLists.txt [hunk 1]
```diff
@@ -2,13 +2,16 @@ cmake_minimum_required(VERSION 3.15)
 project (crow_ssl_test)
 
 include(${CMAKE_SOURCE_DIR}/cmake/compiler_options.cmake)
+include(${CMAKE_SOURCE_DIR}/cmake/CPM.cmake)
+
+CPMAddPackage("gh:catchorg/Catch2@3.6.0")
```
Comment (gittiver) [inline tests/ssl/CMakeLists.txt:7]: Isn't that inclusion superfluous as it is also included one level above?

**Review by gittiver — COMMENTED** [review]
See comments

File: tests/ssl/CMakeLists.txt (hunk 1 above)
Comment (mcparry) [reply tests/ssl/CMakeLists.txt:6]: Yes indeed it was.  Removed in next commit.

Comment (gittiver) [reply tests/ssl/CMakeLists.txt:6]: MAddPackage("gh:catchorg/Catch2@3.6.0") was already done in parent dir /test, so it could be removed here too?

Comment (mcparry) [reply tests/ssl/CMakeLists.txt]: Agreed and removed.

**Review by gittiver — APPROVED** [review]
thx  for your work, looks good for me now.

**Pull request merged**